WIP/calculator.rb

```diff
@@ -0,0 +1,104 @@
+class CheckoutSummary
+
+  attr_reader :discounts
+
+  #the default value for accumulate is true unless specified or set to be false
+  def initialize(**args)
+    @accumulate = args[:accumulate] || args[:accumulate] == nil
+    @discounts = {}
+    @gross_cost = args[:gross_cost]
```

Comment:
what if `@gross_cost` is not passed in? we need some validation somewhere to validate this field?